tests/TestCase.php

```diff
@@ -154,6 +156,25 @@ protected function assertDeprecated(callable $execution): void
         $this->assertCount(1, $errors);
     }
 
+    protected function createOptionDataProvider(array $options): array
+    {
+        $data = [];
+
+        // Loop through each option and get possible values
+        foreach ($options as $option => $values) {
+            // Create an option array with a named data provider for each value
+            foreach ($values as $key => $value) {
+                if (is_int($key)) {
+                    $key = get_debug_type($value);
+                }
+
+                $data[$option . '_' . $key] = [[$option => $value]];
```

Comment:
Does this potentially break if a data provider includes two values with the same type? I suppose PHPUnit would raise an error for that case, so we can leave as-is and address it if it ever becomes an issue down the line.

Reply:
PHPUnit wouldn't raise for that case, but we'd silently ignore tests. Since that is extremely hard to detect and can cause significant problems, I've changed this to use `<option>_<key>_<type>` as data set name for numeric keys, and `<option>_<key>` for string keys. That should remove all possible duplicates.